### ISSUE TITLE
Android Emulator

### ISSUE THREAD

**Issue by shatyuka** [ThatGuySam/doesitarm#395]
**The official name of the app**
Android Emulator

**Is there a supported version available on a stable release channel?**
No

**Proposed App Status**
✅ Yes, Full Native Apple Silicon Support

**Proposed App Category**
Developer Tools

**Related Issue Tracker Link or discussion**
https://androidstudio.googleblog.com/2020/12/android-emulator-apple-silicon-preview.html
https://github.com/741g/android-emulator-m1-preview

**An Official App Download Page**
https://github.com/741g/android-emulator-m1-preview/releases/tag/0.1

**Additional context**
[Android Studio page](https://doesitarm.com/app/android-studio/) also needs an update.
Screenshot:
<img width="558" alt="screenshot" src="https://user-images.githubusercontent.com/31368738/101277242-088af280-37ee-11eb-85c0-e2f54263f893.png">

**Comment by ThatGuySam** [comment]
Thanks for the App Request! 

This has definitely been a big blocker for Android Developers. 

We can't _actually_ list it as fully supported until the supported build is moved to the App's normal stable release channel. 

I have, however, added a link to the Android Studio page. 

We'll leave this issue open until the full stable build is released

**Comment by thejoaov** [comment]
@ThatGuySam 
It is stable by now, the emulator was added to the sdk, so, the emulator preview is no longer needed.
https://github.com/google/android-emulator-m1-preview#note-no-longer-needed

**Comment by shatyuka** [comment]
@thejoaov However, the SDK Manager will download the x64 emulator. And inside it, there is a bash file `darwin-aarch64-replace.sh`. It will download the aarch64 emulator from github and replace the x64 one. That is, still preview yet.
```bash
EMULATOR_DOWNLOAD_BASE_NAME="emulator-darwin-aarch64-0.2"
rm -rf ${EMULATOR_DOWNLOAD_BASE_NAME}
echo "curl -L https://github.com/google/android-emulator-m1-preview/releases/download/0.2/${EMULATOR_DOWNLOAD_BASE_NAME}-engine-only.zip -o ${EMULATOR_DOWNLOAD_BASE_NAME}.zip"
curl -L https://github.com/google/android-emulator-m1-preview/releases/download/0.2/${EMULATOR_DOWNLOAD_BASE_NAME}-engine-only.zip -o ${EMULATOR_DOWNLOAD_BASE_NAME}.zip && \
echo "unzip ${EMULATOR_DOWNLOAD_BASE_NAME}.zip" && \
unzip ${EMULATOR_DOWNLOAD_BASE_NAME}.zip && \
```

**Comment by moffpage** [comment]
Idk the specifics but you can download arm64 compatible Android emulator even on stable Android Studio 4.2.1:
<img width="536" alt="image" src="https://user-images.githubusercontent.com/33549573/123580113-f3fa6000-d7fa-11eb-94aa-c693d8ee5b6a.png">
I have the latest Canary, Beta builds from Google and installed the official one so that this issue can be closed, wanted to prove you don't need a Beta or Canary (which as IDEs have support for Apple Silicon, let's separate the emulator part -> it wouldn't be integrated into the IDE and you would still have to download the 5GB Preview .app from GitHub) to install a natively supported emulator.
<img width="536" alt="image" src="https://user-images.githubusercontent.com/33549573/123580307-5eab9b80-d7fb-11eb-8d68-24399ec69440.png">
This is a screenshot from the stable Android Studio build.

**Comment by shatyuka** [comment]
@moffpage Seems that you are talking about the system image, not the emulator itself.
Also, this image goes ahead of the GitHub one. So you still need preview version of Android Emulator to get ADB functionable.

**Comment by moffpage** [comment]
I don't really understand how do they differ. Do you really need the emulator as a separate app? What does "this image does ahead of the Github one" mean?

**Comment by shatyuka** [comment]
> I don't really understand how do they differ. Do you really need the emulator as a separate app? What does "this image does ahead of the Github one" mean?

No, separate app is not needed. As described above, for stable version, there is a bash file exist in the emulator directory. Execute it and you will get an aarch64 emulator downloaded form github. But it is outdated and not compatible with newer system images (see [this](https://stackoverflow.com/questions/67230200/emulator-appearing-offline-on-m1-mac-after-the-last-update-of-arm64-v8a)). You still need the preview version of emulator to get it work.

**Comment by moffpage** [comment]
Please remove "Waiting for Stable Release", it's already built in Android Studio Beta

**Comment by moffpage** [comment]
By that I meant also update the website and close the issue 😂

**Comment by ThatGuySam** [comment]
Thanks @shatyuka for the request and follow-ups. Published and verified: [Android Emulator](https://doesitarm.com/app/android-emulator) and the corrected [Android Studio listing](https://doesitarm.com/app/android-studio).

[Google's stable Arctic Fox release notes](https://developer.android.com/studio/releases/past-releases/as-arctic-fox-release-notes#apple-silicon) cover the ARM64 host applications and corresponding emulator images. The listing no longer says native support is preview-only. This does not imply that every SDK/NDK tool or x86 guest image runs natively. I also removed the unrelated BrickLink Studio scan association and added a regression test.

Ready for closure after maintainer review. Please keep suggesting developer tools we're missing.
